text/0000-explode.md

```diff
@@ -0,0 +1,242 @@
+
+We propose adding an *additional* supported way to depend on Ember. You can choose to run `ember explode`, and we will remove `ember-source` from your dependencies and replace it with `@ember/core` *plus the current complete set of already-extracted Ember packages*. `ember explode` is intended to not change the set of Ember features in your app. It just splats them out into a form that comes from separate packages.
+
+`@ember/core` and all of the packages split out from it will share a lockstep version number. We will manage releases of these packages in two phases:
+
+ - initial phase: while we are still establishing the package boundaries, we will start at 0.1.0 and signal breaking releases each time we split out a new package.
+ - stabilization: once we have created a complete set of packages and stabilized their boundaries, we will update the version number to match `ember-source`, after which the split-up family of packages and the aggregated `ember-source` packages will stay in lock-step.
+
+## Release Cadence
+
+This proposal doesn't change Ember's release cadence. The new packages should be released on the same schedule as `ember-source` (and of course they contain the same code -- as long as you install the full set of them).
+
+Even though we expect to have many "breaking" releases of `@ember/core` during the initial phase, they will only "break" in the sense of requiring apps to add additional dependencies on new split-out packages. We can extend the `ember-cli-update` tool to handle this case automatically, making it easier for early adopters of the new package format to follow along.
+
+## Inter-package dependencies
+
+If one of the newly-split-off Ember packages depends on another, it will say so in its NPM peerDependencies. We will use exact-version constraints, which is effectively how things already work today.
+
+NPM has historically been loose about peerDependencies, so they are often ignored by developers. We propose that ember-cli should hard error for missing peerDependencies only in Ember's own packages to avoid this problem (most people who think they don't really need to clean up peerDependency warnings are simply mistaken, and have latent bugs).
+
+An environment variable will be created to suppress the error so that people who are doing advanced debugging of the ember packages themselves can  test scenarios that involve mismatched package versions.
+
+## Addon dependencies on Ember packages
+
+Right now, every addon in the ecosystem is allowed to implicitly depend on any feature in Ember. You can't know *a priori* which features are safe to remove if you're using any given addon.
+
+Therefore, we propose new metadata in the `ember-addon` section of `package.json`:
+
+```js
+  "ember-addon": {
+    "required-core-features": "0.4.0",
+    "required-additional-features": ["@ember/routing", "@ember/prototype-etxensions", "@ember/string"]
+  }
```

Comment:
Just flagging that there's going to be some amount of overlap with deprecation staging config here #649. We discussed making it possible to identify "parts of ember they do not use", but decided against it. Nothing really to do at the moment, since it's not really feasible to try to expand the scope of both RFCs and there isn't anything inherently conflicting yet. Just wanted to let you know. cc @pzuraq @rwjblue.